File: Chain of Responsibility/Golang/Authentication.go
package main

import (
	"fmt"
)

type Authentication struct {
	nextHandler Handler
}

func (auth *Authentication) setNextHandler(handler Handler) {
	auth.nextHandler = handler
}

func (auth *Authentication) execute(request *Request) *Response  {
	if request.Authenticated {
		fmt.Println("Authenticated")

		return auth.nextHandler.execute(request)
	} else {
		fmt.Println("Unauthenticated")

		return &Response{
			Code: 		403,
			Message: 	"Unauthenticated",
		}
	}
}

File: Chain of Responsibility/Ruby/Authentication.rb
require_relative "Handler"
require_relative "Response"

class Authentication < Handler
    def execute(request)
        if request.authenticated
            puts "Authenticated"
            return @next_handler.execute(request)
        else
            return Response.new(
                code: 403, 
                message: "Unauthenticated",
                data: nil
            )
        end
    end
end
